smoke-tests/src/test/java/com/splunk/opentelemetry/AppServerTest.java

```diff
@@ -132,24 +134,25 @@ protected void assertServerHandler(ExpectedServerAttributes serverAttributes)
     Request request = new Request.Builder().get().url(url).build();
     Response response = client.newCall(request).execute();
     log.debug("Response for non-existing page: {}", response.body().string());
-    Assertions.assertEquals(
+    assertEquals(
         404,
         response.code(),
         "404 response code is expected from the app-server for a request to a non-existing page.");
     var traces = waitForTraces();
 
-    Assertions.assertEquals(1, traces.size(), "There is one trace from server handler");
+    var numberOfTraceIds = traces.getSpanStream().map(Span::getTraceId).distinct().count();
```

Comment:
⭐ This is the important change.  Depending on how the data is reported, `traces.size()` can be 1 or 2, even though both spans are always for a single trace (same ID).

Reply:
There is `com.splunk.opentelemetry.TraceInspector#getTraceIds` which does the part up to `distinct`. Maybe we can have just one way of doing this?

Reply:
I consider it to be a different use case.  Here, we are only concerned with the number of distinct IDs, but with `getTraceIds()` the trace ID values are actually used/compared.  I guess since you called it out here _and_ Nikita also got tripped up below, I'll combine them.